### PR TITLE
Extend logging functionality

### LINKED ISSUE #8: Extend bridge logging functionality
It is necessary to extend the logging functionality in order to have information handled and generated by the bridge instance more transparent.
Below is the list of information needed for every type of operation:

1. `processDeposits`
   * tx hash of the transaction (which emitted `Deposit()` event on  _HomeBridge_ contract) including the transaction sender and value;
   * tx hash of the transaction sent by the bridge instance to  _HomeBridge_ contract to approve the deposit including a reference tx hash originated the deposit (it allows to link this transaction to transaction from previous item);
   * information whether `SignedForDeposit()` emitted by transaction from the previous item including the reference tx hash originated the deposit; 
   * tx hashes for transactions which demonstrate that the deposit was handled by other instances (other `SignedForDeposit()` events) with the reference tx hash originated the deposit.

2. `processCollectedSignatures`
   * tx hash of the transaction (which emitted `CollectedSignatures()` event) including information if this bridge instance is responsible for relaying the deposit message;
   * tx hash of the transaction sent by the bridge instance to the _ForeignBridge_ contract including a reference tx hash originated the deposit (it allows to link this transaction to transaction from previous item);
   * information whether respond from _ForeignBridge_ contract (`Deposit()` event) received including the reference tx hash originated the deposit;
   * tx hashes for transaction which demonstrate that the deposit was relayed by other instances (other `Deposit()` events) with the reference tx hash originated the deposit.

3. `processWithdraw`
   * tx hash of the transaction (which emitted `Withdraw()` event on _ForeignBridge_ contract) including the transaction sender and value;
   * tx hash of the transaction sent by the bridge instance to _HomeBridge_ contract to approve withdrawal including a reference tx hash originated the withdrawal (it allows to link this transaction to transaction from previous item);
   * information whether `Withdraw()` emitted by transaction from the previous item including the reference tx hash originated the withdrawal; 
   * tx hashes for transactions which demonstrate that the withdrawal was handled by other instances (other `Withdraw()` events) with the reference tx hash originated the withdrawal.

### PR DESCRIPTION
**Pull request by patitonar**
Closes #8 

Added [Pino](https://github.com/pinojs/pino/) for structured logs.
If the process is running in production mode (env variable `NODE_ENV=production`) the log output will be a JSON line, otherwise the output will be prettified for development purposes.

### Example logs
Production Mode

```
{"level":30,"time":1531747352923,"msg":"Connected to redis","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"watcher-deposit","v":1}
{"level":30,"time":1531747352985,"msg":"Connected to amqp Broker","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"watcher-deposit","v":1}
{"level":30,"time":1531747355522,"msg":"Found 0 Deposit events","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"watcher-deposit","v":1}
{"level":30,"time":1531747375891,"msg":"Found 1 Deposit events","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"watcher-deposit","v":1}
{"level":30,"time":1531747375894,"msg":"Processing Deposit 0x7d4a02d513ef3129ce0829987de6280ff07a74258908df7534123117e4a343ac","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"watcher-deposit","eventTransactionHash":"0x7d4a02d513ef3129ce0829987de6280ff07a74258908df7534123117e4a343ac","sender":"0xf4BEF13F9f4f2B203FAF0C3cBbaAbe1afE056955","value":"10000000000000000","v":1}
{"level":30,"time":1531747376621,"msg":"Transactions to send: 1","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"watcher-deposit","v":1}
```

```
{"level":40,"time":1531747971659,"msg":"You are using http (http://sokol.poa.network). In production https must be used instead.","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"watcher-deposit","v":1}
{"level":40,"time":1531747971660,"msg":"You are using http (http://kovan.infura.io/mew). In production https must be used instead.","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"watcher-deposit","v":1}
{"level":30,"time":1531747971685,"msg":"Connected to redis","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"watcher-deposit","v":1}
{"level":30,"time":1531747971736,"msg":"Connected to amqp Broker","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"watcher-deposit","v":1}
{"level":50,"time":1531747973791,"msg":"Required block confirmations cannot be obtained","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"watcher-deposit","type":"Error","stack":"Error: Required block confirmations cannot be obtained\n    at getRequiredBlockConfirmations (/home/altoros/repos/bridge-nodejs/src/tx/web3.js:29:11)\n    at <anonymous>","v":1}
{"level":50,"time":1531747975680,"msg":"Block Number cannot be obtained","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"watcher-deposit","type":"Error","stack":"Error: Block Number cannot be obtained\n    at getBlockNumber (/home/altoros/repos/bridge-nodejs/src/tx/web3.js:13:11)\n    at <anonymous>","v":1}
```
```
{"level":30,"time":1531748228337,"msg":"Connected to redis","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"sender-home","v":1}
{"level":30,"time":1531748228390,"msg":"Connected to amqp Broker","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"sender-home","v":1}
{"level":30,"time":1531748229085,"msg":"Msg received with 1 Tx to send","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"sender-home","v":1}
{"level":30,"time":1531748231019,"msg":"Tx generated 0x994f2f625ea23a65198da14a936cb7e54d7afe7baa1807d01c5978d11a7c17ea for event Tx 0x94140653c354703a76df567e55ac6eb0c5315e77564d484497759b3f3d977b0b","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"sender-home","eventTransactionHash":"0x94140653c354703a76df567e55ac6eb0c5315e77564d484497759b3f3d977b0b","generatedTransactionHash":"0x994f2f625ea23a65198da14a936cb7e54d7afe7baa1807d01c5978d11a7c17ea","v":1}
{"level":30,"time":1531748231023,"msg":"Finished processing msg","validator":"0x3752F0a521b2E383693f658D6FC377953024C207","name":"sender-home","v":1}
```

Development Mode
watcher
watchererror
sender

### CODE REVIEW

**Comment by patitonar** [comment]
@phahulin I don't know how are we currently sending the logs to papertrail on the rust version, but something that can help is to use [pino-socket](https://github.com/pinojs/pino-socket)
Example of use when running a process: 
`npm run watcher:deposit | pino-socket -a logs.papertrailapp.com -p xxxxx`

**Comment by phahulin** [comment]
@patitonar in current version, bridge is started as a service (via systemd), its logs are written to linux syslog. syslog is configured to send all logs to papertrail. So it's not done by the bridge service itself

File: src/events/processCollectedSignatures.js [hunk 1]
```diff
@@ -53,7 +58,10 @@ async function processCollectedSignatures(signatures) {
       try {
         gasEstimate = await foreignBridge.methods.deposit(v, r, s, message).estimateGas()
       } catch (e) {
-        console.log(indexSig + 1, ' # already processed col sig', colSignature.transactionHash)
+        logger.info(
+          { eventTransactionHash: colSignature.transactionHash },
+          `Already processed CollectedSignatures ${colSignature.transactionHash}`
```
[the remark this answers is not in the document]
Comment (patitonar) [reply src/events/processCollectedSignatures.js]: I agree, I think we should add a check for RPC connection lost. For other cases of fail, I think they should be discussed and solved as part of #24

**Comment by patitonar** [comment]
@akolotov Added a specific handle for RPC connection errors on estimageGas to avoid getting the message `Already processed event` in that case.

**Comment by patitonar** [comment]
@rstormsf solved merge conflicts. Updated tests

**Pull request merged**